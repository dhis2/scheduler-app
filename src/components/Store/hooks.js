import { useContext } from 'react'
import StoreContext from './StoreContext'

/**
 * The state for the job filter and showing system
 * jobs is used in the job list, but kept in the
 * store since it has to persist after a refetch.
 */

export const useJobFilter = () => {
    const store = useContext(StoreContext)
    return store.jobFilter
}

export const useShowSystemJobs = () => {
    const store = useContext(StoreContext)
    return store.showSystemJobs
}
