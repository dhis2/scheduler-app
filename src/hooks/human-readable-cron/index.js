export { default as useHumanReadableCron } from './use-human-readable-cron'
