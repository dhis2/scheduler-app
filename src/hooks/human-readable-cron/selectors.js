export const getLocale = data => {
    const { keyUiLocale } = data

    if (!keyUiLocale) {
        return ''
    }

    return keyUiLocale
}
