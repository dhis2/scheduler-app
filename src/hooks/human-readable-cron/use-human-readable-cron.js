import cronstrue from 'cronstrue/i18n'
import { useContext } from 'react'
import { validateCron } from '../../services/validators'
import { LocaleContext } from '../../components/Context'

const useHumanReadableCron = cronExpression => {
    const locale = useContext(LocaleContext)
    const isValid = cronExpression && validateCron(cronExpression)

    if (!locale || !isValid) {
        return ''
    }

    return cronstrue.toString(cronExpression, { locale })
}

export default useHumanReadableCron
