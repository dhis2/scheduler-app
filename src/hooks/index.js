export * from './jobs'
export * from './job-schedules'
export * from './job-types'
export * from './parameter-options'
