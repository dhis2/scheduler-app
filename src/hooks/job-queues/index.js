export { default as useSubmitJobQueue } from './use-submit-job-queue'
