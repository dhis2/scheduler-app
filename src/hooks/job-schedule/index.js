import useJobSchedules from './use-job-schedules'

export { useJobSchedules }
