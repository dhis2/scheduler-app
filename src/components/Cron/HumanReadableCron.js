import { string } from '@dhis2/prop-types'
import cronstrue from 'cronstrue/i18n'
import { useGetUserSettings, selectors } from '../../hooks/user-settings'

const HumanReadableCron = ({ cronExpression }) => {
    const { loading, error, data } = useGetUserSettings()

    if (loading) {
        return null
    }

    // Fall back to default locale in case of errors (English for cronstrue)
    if (error) {
        return cronstrue.toString(cronExpression)
    }

    const locale = selectors.getLocale(data)

    return cronstrue.toString(cronExpression, { locale })
}

HumanReadableCron.propTypes = {
    cronExpression: string.isRequired,
}

export default HumanReadableCron
