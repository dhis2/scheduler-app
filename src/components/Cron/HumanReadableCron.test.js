import { useHumanReadableCron } from '../../hooks/human-readable-cron'
import HumanReadableCron from './HumanReadableCron'

jest.mock('../../hooks/human-readable-cron', () => ({
    useHumanReadableCron: jest.fn(),
}))

afterEach(() => {
    jest.resetAllMocks()
})

describe('<HumanReadableCron>', () => {
    it('returns a human readable cron', () => {
        const cronExpression = '0 0 1 ? * *'
        const humanReadableCron = 'Every day'

        useHumanReadableCron.mockImplementation(() => humanReadableCron)

        expect(HumanReadableCron({ cronExpression })).toBe(humanReadableCron)
    })
})
