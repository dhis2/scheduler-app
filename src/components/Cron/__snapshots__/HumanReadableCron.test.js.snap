// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HumanReadableCron> returns nothing while loading 1`] = `""`;
